kpatch.py

```diff
@@ -132,63 +164,91 @@ def _read_conf(self):
 
         try:
-            with open(self.cfg_file, 'w') as cfg_stream:
+            with open(self.cfg_file, 'w', encoding='utf-8') as cfg_stream:
                 conf.write(cfg_stream)
         except Exception as e:
             raise dnf.exceptions.Error(_("Failed to update conf file: {}").format(str(e)))
 
 
     def run(self):
+        """
+        Decision tree, execution based on config
+        """
         action = self.opts.action
 
-        if action == "auto":
+        if action in ("auto-update", "auto"):
             self._install_missing_kpp_pkgs()
-            self._update_plugin_cfg(True)
-        elif action == "manual":
-            self._update_plugin_cfg(False)
+            self._update_plugin_cfg(KPATCH_UPDATE_OPT, True)
+            logger.info(_("Kpatch update setting: {}").format(action))
+
+        elif action in ("manual-update", "manual"):
+            self._update_plugin_cfg(KPATCH_UPDATE_OPT, False)
+            logger.info(_("Kpatch update setting: {}").format(action))
+
+        elif action == "auto-filter":
+            self._update_plugin_cfg(KPATCH_FILTER_OPT, True)
+            logger.info(_("Kpatch filter setting: {}").format(action))
+
+        elif action == "no-filter":
+            self._update_plugin_cfg(KPATCH_FILTER_OPT, False)
+            logger.info(_("Kpatch filter setting: {}").format(action))
+
         elif action == "status":
             conf = self._read_conf()
-            kp_status = "manual"
+            kp_status = "manual-update"
             if (conf is not None and conf.has_section('main') and
                 conf.has_option('main', KPATCH_UPDATE_OPT) and
                 conf.getboolean('main', KPATCH_UPDATE_OPT)):
-                kp_status = "auto"
+                kp_status = "auto-update"
             logger.info(_("Kpatch update setting: {}").format(kp_status))
 
+            kp_status = "no-filter"
+            if (conf is not None and conf.has_section('main') and
+                conf.has_option('main', KPATCH_FILTER_OPT) and
+                conf.getboolean('main', KPATCH_FILTER_OPT)):
+                kp_status = "auto-filter"
+            logger.info(_("Kpatch filter setting: {}").format(kp_status))
+
             kpps = self._list_missing_kpp_pkgs()
             if kpps:
                 logger.info(_("Available patches: {}").format(", ".join(kpps)))
 
         elif action == "install":
             self._install_missing_kpp_pkgs()
+
         else:
             raise dnf.exceptions.Error(_("Invalid argument: {}").format(action))
 
 
-
 class KpatchPlugin(dnf.Plugin):
+    """ kpatch specific plugin, that helps to filter kpatch supported kernels """
 
     name = KPATCH_PLUGIN_NAME
 
+    # list of package names to filter based on kpatch support
+    kernel_pkg_names = ['kernel', 'kernel-core', 'kernel-modules',
+                        'kernel-modules-core', 'kernel-modules-extra']
+    kpatch_requirement = ['kernel', 'kernel-uname-r']
```

Comment:
Placeholder TODO, we should find out if we should be filtering any realtime kernel package variants as well.